### PR TITLE
fix: ArgParse's registry for the type keyword argument to add_argument

### PR DESCRIPTION
**Pull request by Mr-Sunglasses**
fix: #11034

### CODE REVIEW

**Comment by srittau** [comment]
Note: #11034 is about the type registry, which would require adding a `str` argument, which this PR doesn't seem to fix.

File: stdlib/argparse.pyi [hunk 1]
```diff
@@ -32,6 +32,7 @@ _T = TypeVar("_T")
 _ActionT = TypeVar("_ActionT", bound=Action)
 _ArgumentParserT = TypeVar("_ArgumentParserT", bound=ArgumentParser)
 _N = TypeVar("_N")
+_ActionType: TypeAlias = Callable[[str], Any] | type[FileType]
```
Comment (srittau) [inline stdlib/argparse.pyi:35]: I'm not sure `type[FileType]` is correct. At least [the documentation](https://docs.python.org/3/library/argparse.html#type) uses examples like this:

```python
parser.add_argument('dest_file', type=argparse.FileType('w', encoding='latin-1'))
```

[And says](https://docs.python.org/3/library/argparse.html#argparse.FileType):

> The [FileType](https://docs.python.org/3/library/argparse.html#argparse.FileType) factory creates objects that can be passed to the type argument of [ArgumentParser.add_argument()](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument).

Comment (Mr-Sunglasses) [reply stdlib/argparse.pyi:35]: Thanks, @srittau , I went through the docs and You're correct. The `argparse.FileType` is actually a factory that returns a callable object suitable for use as the `type` argument in `add_argument`. It doesn't make sense to use `type[FileType]`

**Review by srittau — APPROVED** [review]
Thanks!

**Pull request merged**